pkg/common/cns-lib/volume/manager.go

```diff
@@ -2112,6 +2112,56 @@ func (m *defaultManager) deleteSnapshotWithImprovedIdempotencyCheck(
 	// Get the taskInfo
 	deleteSnapshotsTaskInfo, err := cns.GetTaskInfo(ctx, deleteSnapshotTask)
 	if err != nil || deleteSnapshotsTaskInfo == nil {
+		if err != nil && cnsvsphere.IsManagedObjectNotFound(err, deleteSnapshotTask.Reference()) {
```

Comment:
The errors from the two bugs look a little different.  One says "with error TaskInfo result is empty", and the other one says "with error Failed to get taskInfo for DeleteSnapshots task from vCenter xxxxxx with err: ServerFaultCode: The object 'vim.Task:task-247323' has already been deleted or has not been completely created".
Does this check cover both cases?